afs-security-analysis-local/src/test/java/com/powsybl/afs/security/local/LocalSecurityAnalysisRunningServiceTest.java

```diff
@@ -9,51 +9,17 @@
 import com.google.common.collect.ImmutableList;
 import com.powsybl.afs.ServiceExtension;
 import com.powsybl.afs.ext.base.LocalNetworkCacheServiceExtension;
-import com.powsybl.computation.ComputationManager;
-import com.powsybl.contingency.ContingenciesProvider;
-import com.powsybl.iidm.network.Network;
-import com.powsybl.security.*;
 import com.powsybl.afs.security.SecurityAnalysisRunnerTest;
-import com.powsybl.security.interceptors.SecurityAnalysisInterceptor;
 
-import java.util.Collections;
 import java.util.List;
-import java.util.concurrent.CompletableFuture;
 
 /**
  * @author Sylvain Leclerc <sylvain.leclerc at rte-france.com>
  */
 public class LocalSecurityAnalysisRunningServiceTest extends SecurityAnalysisRunnerTest {
 
-    private static class SecurityAnalysisFactoryMock implements SecurityAnalysisFactory {
-        @Override
-        public SecurityAnalysis create(Network network, ComputationManager computationManager, int priority) {
-            return new SecurityAnalysis() {
-
-                @Override
-                public void addInterceptor(SecurityAnalysisInterceptor interceptor) {
-                    // Not implemented
-                }
-
-                @Override
-                public boolean removeInterceptor(SecurityAnalysisInterceptor interceptor) {
-                    // Not implemented
-                    return false;
-                }
-
-                @Override
-                public CompletableFuture<SecurityAnalysisResult> run(String workingStateId, SecurityAnalysisParameters securityAnalysisParameters, ContingenciesProvider contingenciesProvider) {
-                    LimitViolationsResult preContingencyResult = new LimitViolationsResult(true, ImmutableList.of(new LimitViolation("s1", LimitViolationType.HIGH_VOLTAGE, 400.0, 1f, 440.0)));
-                    SecurityAnalysisResult result = new SecurityAnalysisResult(preContingencyResult, Collections.emptyList());
-                    return CompletableFuture.completedFuture(result);
-                }
-            };
-        }
-    }
-
-    @Override
     protected List<ServiceExtension> getServiceExtensions() {
-        return ImmutableList.of(new LocalSecurityAnalysisRunningServiceExtension(SecurityAnalysisFactoryMock::new),
+        return ImmutableList.of(new LocalSecurityAnalysisRunningServiceExtension(),
```

Comment:
Here it would be simpler to provide the `SecurityAnalysisProvider` in the constructor, instead of using an `@AutoService` extension.